FireRender.Maya.Src/Context/HybridContext.cpp

```diff
@@ -97,13 +97,23 @@ void HybridContext::setupContextPostSceneCreation(const FireRenderGlobalsData& f
 	setSamplesPerUpdate(1);
 
 	int maxRecursion = 0;
+	bool isRestir = false;
 	if (GetRenderType() == RenderType::ProductionRender) // production (final) rendering
 	{
 		maxRecursion = fireRenderGlobalsData.maxRayDepth;
+		isRestir = fireRenderGlobalsData.productionReservoirSampling != 0;
 	}
 	else if (isInteractive())
 	{
 		maxRecursion = fireRenderGlobalsData.viewportMaxRayDepth;
+		isRestir = fireRenderGlobalsData.viewportReservoirSampling != 0;
+	}
+
+	// Restir with large number RPR_CONTEXT_MAX_RECURSION ~ 15 will require a huge buffers for the Restir needs (around 23Gb) to reduce consumption
```

Comment:
Comment says Max_Recursion of 15 but the check is for > 5. Which causes the issue? 

IMO most renders make sense at 5-10 ray depth.  If we can limit to 10 that might make sense

Reply:
Comment says Max_Recursion of 15 since in the trace that was sent to Hybrid Max_Recursion  was set to 15 and they said it was too much. The comment is a mostly copied reply from Jira.

Reply:
Do we have any word from them what max depth is not "too much"?  Is 10 ok?

Reply:
Here is the link
https://amdrender.atlassian.net/browse/RPRHYB-941

Reply:
Ok sorry.  In conversation with them we just want to set maxRecursion to 5.  both in the comment and code.